packages/algorand-js/src/runtime/account.ts

```diff
@@ -120,9 +103,9 @@ export class StoreAccount implements StoreAccountI {
         throw new Error('Maximum Opt In applications per account is 10');
       }
 
-      const localParams: AppLocalState = {
+      const localParams: AppLocalStateM = {
         id: appId,
-        "key-value": [],
+        "key-value": new Map<string, StackElem>(),
         schema: appParams["local-state-schema"]
       };
```

Comment:
what if we want to update the application ? does that mean we need to reset the local-state of opt-in accounts ?

Reply:
Good point. I haven't tested that, will test it to check if local state is reset on update or not.

Reply:
Upon confirming: no, the state is not resetted. And in this function we are already checking previously if app is opted in or not. If app is not opted in, then only we are creating a new local state and setting. Please let me know if that makes sense.